projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/post-fields-publicize-connections.php

```diff
@@ -102,64 +104,40 @@ public function get_schema() {
+					__( 'Deprecated in favor of %s.', 'jetpack' ),
+					'connection_id'
+				),
+			),
+			'username' => array(
+				'type'        => 'string',
+				'description' => __( 'Username of the connected account.', 'jetpack' ) . ' ' . sprintf(
+					/* translators: %s is the new field name */
+					__( 'Deprecated in favor of %s.', 'jetpack' ),
+					'external_handle'
+				),
+			),
+		);
+
 		return array(
 			'$schema'    => 'http://json-schema.org/draft-04/schema#',
 			'title'      => 'jetpack-publicize-post-connection',
 			'type'       => 'object',
-			'properties' => array(
-				'id'              => array(
-					'description' => __( 'Unique identifier for the Jetpack Social connection', 'jetpack' ),
-					'type'        => 'string',
-					'context'     => array( 'view', 'edit' ),
-					'readonly'    => true,
-				),
-				'service_name'    => array(
-					'description' => __( 'Alphanumeric identifier for the Jetpack Social service', 'jetpack' ),
-					'type'        => 'string',
-					'context'     => array( 'view', 'edit' ),
-					'readonly'    => true,
-				),
-				'display_name'    => array(
-					'description' => __( 'Display name of the connected account', 'jetpack' ),
-					'type'        => 'string',
-					'context'     => array( 'view', 'edit' ),
-					'readonly'    => true,
-				),
-				'username'        => array(
-					'description' => __( 'Username of the connected account', 'jetpack' ),
-					'type'        => 'string',
-					'context'     => array( 'view', 'edit' ),
-					'readonly'    => true,
-				),
-				'profile_picture' => array(
-					'description' => __( 'Profile picture of the connected account', 'jetpack' ),
-					'type'        => 'string',
-					'context'     => array( 'view', 'edit' ),
-					'readonly'    => true,
-				),
-				'enabled'         => array(
-					'description' => __( 'Whether to share to this connection', 'jetpack' ),
-					'type'        => 'boolean',
-					'context'     => array( 'edit' ),
-				),
-				'done'            => array(
-					'description' => __( 'Whether Jetpack Social has already finished sharing for this post', 'jetpack' ),
-					'type'        => 'boolean',
-					'context'     => array( 'edit' ),
-					'readonly'    => true,
-				),
-				'toggleable'      => array(
-					'description' => __( 'Whether `enable` can be changed for this post/connection', 'jetpack' ),
-					'type'        => 'boolean',
-					'context'     => array( 'edit' ),
-					'readonly'    => true,
-				),
-				'external_id'     => array(
-					'description' => __( 'The external ID of the connected account', 'jetpack' ),
-					'type'        => 'string',
-					'context'     => array( 'view', 'edit' ),
-					'readonly'    => true,
-				),
+			'properties' => array_merge(
+				$deprecated_fields,
+				$connection_fields,
+				array(
+					'enabled' => array(
+						'description' => __( 'Whether to share to this connection.', 'jetpack' ),
+						'type'        => 'boolean',
+						'context'     => array( 'edit' ),
+					),
+				)
```

Comment:
These schema changes are necessary to ensure that the fields are as per our schema because the fields are filtered below based on the schema.